### PR TITLE
minor: Port more window tests to sqlogictests 

### PR DESCRIPTION
**Pull request by alamb**
# Which issue does this PR close?

Part of https://github.com/apache/arrow-datafusion/issues/4495

Follow on to https://github.com/apache/arrow-datafusion/pull/5330

# Rationale for this change

I am trying to keep the testability of DataFusion reasonable. Sqllogictests are easier to add / update so we are trying to move stuff over there.

# What changes are included in this PR?

Port more window tests to window.slt

# Are these changes tested?
Yes all tests

# Are there any user-facing changes?
No

### CODE REVIEW

File: datafusion/core/tests/sql/window.rs [hunk 1]
```diff
@@ -61,912 +61,6 @@ async fn window_frame_creation_type_checking() -> Result<()> {
...
-        "| -1114      | -1927628110 |",
-        "| 15673      | -1899175111 |",
-        "+------------+-------------+",
-    ];
-    assert_batches_eq!(expected, &actual);
-
-    Ok(())
-}
-
 fn write_test_data_to_parquet(tmpdir: &TempDir, n_file: usize) -> Result<()> {
```
Comment (alamb) [inline datafusion/core/tests/sql/window.rs:64]: this file has only a few hundred lines after this one 😅

**Comment by tustvold** [comment]
There appear to be failing tests?

**Comment by alamb** [comment]
> 
There appear to be failing tests?



Looks like some non deterministic output -- will fix

**Review by jackwener — APPROVED** [review]

File: datafusion/core/tests/sqllogictests/test_files/window.slt [hunk 2]
```diff
@@ -92,7 +92,7 @@ limit 5
 
 
 
-# async fn csv_query_window_with_order_by
```
[the remark this answers is not in the document]
Comment (alamb) [reply datafusion/core/tests/sqllogictests/test_files/window.slt:95]: If you think it adds value I can put them back in 😆  -- I think it was just left over from incomplete porting

**Pull request merged**